packages/cryptobox/src/main/Cryptobox.ts

```diff
@@ -418,6 +419,85 @@ class Cryptobox extends EventEmitter {
       );
     });
   }
+
+  private async deleteData(): Promise<void> {
+    this.cachedSessions = new LRUCache(DEFAULT_CAPACITY);
+    this.identity = undefined;
+    this.lastResortPreKey = undefined;
+    this.queues = new LRUCache<PriorityQueue>(DEFAULT_CAPACITY);
+    await this.store.delete_all();
+  }
+
+  private async importIdentity(payload: string): Promise<void> {
+    this.logger.log(`Importing local identity...`);
+
+    const identityBuffer = Decoder.fromBase64(payload).asBytes.buffer;
+    const identity = ProteusKeys.IdentityKeyPair.deserialise(identityBuffer);
+
+    await this.save_identity(identity);
+  }
+
+  private async importPreKeys(payload: string[]): Promise<void> {
+    this.logger.log(`Importing "${payload.length}" PreKeys...`);
+
+    const proteusPreKeys = payload.map(preKey => {
+      const preKeyBuffer = Decoder.fromBase64(preKey).asBytes.buffer;
+      const proteusPreKey = ProteusKeys.PreKey.deserialise(preKeyBuffer);
+      if (proteusPreKey.key_id === ProteusKeys.PreKey.MAX_PREKEY_ID) {
+        this.lastResortPreKey = proteusPreKey;
+      }
+      return proteusPreKey;
+    });
+
+    await this.store.save_prekeys(proteusPreKeys);
+  }
+
+  private async importSessions(payload: string[]): Promise<void> {
+    this.logger.log(`Importing "${payload.length}" sessions...`);
+
+    const proteusSessions = payload.map(session => {
+      const sessionBuffer = Decoder.fromBase64(session).asBytes.buffer;
+      return ProteusSession.Session.deserialise(this.identity!, sessionBuffer);
+    });
+
+    const saveSessions = proteusSessions.map(session => {
+      const cryptoBoxSession = new CryptoboxSession(undefined, this.store, session);
+      return this.session_save(cryptoBoxSession);
+    });
+
+    await Promise.all(saveSessions);
+  }
+
+  public async deserialize(payload: SerializedCryptobox) {
+    await this.deleteData();
+    await this.importIdentity(payload.identity);
+    await this.importPreKeys(payload.prekeys);
+    await this.importSessions(payload.sessions);
+    await this.refill_prekeys(true);
+  }
+
+  public async serialize(): Promise<SerializedCryptobox> {
+    const toBase64 = (array: ArrayBuffer) => Encoder.toBase64(new Uint8Array(array)).asString;
```

Comment:
Why do we need `new Uint8Array(array)`? bazinga64's `toBase64()` [also accepts `ArrayBuffer`](https://github.com/wireapp/wire-web-packages/blob/master/packages/bazinga64/src/main/bazinga64.ts#L195).